api/api.proto

```diff
@@ -1014,3 +1014,121 @@ message ClusterAlertEraseRequest {
 
 message ClusterAlertEraseResponse {
 }
+
+// ClusterPairCreateRequest
+// swagger:model
+message ClusterPairCreateRequest {
+    // IP of the remote cluster
+    string remote_cluster_ip = 1;
+    // Port for the remote cluster
+    uint32 remote_cluster_port = 2;
+    // Token used to authenticate with the remote cluster
+    string remote_cluster_token = 3;
+    // Set the new pair as the default
+    bool set_default = 4;
+}
+
+// ClusterPairCreateResponse
+// swagger:response
+message ClusterPairCreateResponse {
+    // ID of the remote cluster
+    string remote_cluster_id = 1;
+    // Name of the remote cluster
+    string remote_cluster_name = 2;
+}
+
+// ClusterPairProcessRequest
+// swagger:model
+message ClusterPairProcessRequest {
+    // ID of the cluster requesting the pairing
+    string source_cluster_id = 1;
+    // Token used to authenticate with the remote cluster
+    string remote_cluster_token = 2;
+}
+
+// ClusterPairProcessResponse
+// swagger:response
+message ClusterPairProcessResponse {
+    //
+    // in: body
+    // ID of the cluster which processed the pair request
+    string remote_cluster_id = 1;
+    // Name of the cluster which processed the pair request
+    //
+    // in: body
+    string remote_cluster_name = 2;
+    // Key/value pair of options returned on successful pairing
+    //
+    // in: body
+    map<string, string> options = 3;
+}
+
+// ClusterPairDeleteRequest
+// swagger:model
+message ClusterPairDeleteRequest {
+    // ID of the cluster pair to be deleted
+    string cluster_id = 1;
```

Comment:
All the clusters paired are remote, so it is kind of redundant. The structs above where I have used remote_cluster_id is where the APIs are being sent between clusters to make it clear.

Reply:
The comment already mentions that. Also, the request is to "Delete" a "ClusterPair", so it implies that it's deleting the pairing with a remote cluster. Adding remote to all the names is redundant unless the APIs are between clusters to avoid confusion.